torch/distributed/_state_dict_utils.py

```diff
@@ -411,17 +411,17 @@ def tensor_func(
             if pin_memory:
 
                 def unpin_memory(t):
-                    succ = torch.cuda.cudart().cudaHostUnregister(t.data_ptr())
+                    succ = int(torch.cuda.cudart().cudaHostUnregister(t.data_ptr()))
                     assert (
```

Comment:
I'm curious to know whether torch.cuda.check_error would be an option to check for errors here.

Reply:
Good catch! I think check error would be reasonable but it currently doesn't support attaching metadata (the assertion string I think is useful).

Maybe we can circle back and extend `check_error`